### PR TITLE
11.0 fix barcode various arm

### PR DESCRIPTION
**Pull request by amoyaux**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/barcodes/static/src/js/barcode_form_view.js [hunk 1]
```diff
@@ -355,9 +357,9 @@ FormController.include({
             title: _t('Set quantity'),
             buttons: [{text: _t('Select'), classes: 'btn-primary', close: true, click: function () {
                 var new_qty = this.$content.find('.o_set_qty_input').val();
-                var values = {};
-                values[activeBarcode.quantity] = parseFloat(new_qty);
-                return self.model.notifyChanges(activeBarcode.candidate.id, values).then(function () {
+                var record = self.model.get(self.handle);
+                return self._barcodeSelectedCandidate(activeBarcode.candidate, record, self.current_barcode, activeBarcode, parseFloat(new_qty))
```
Comment (adr-odoo) [inline addons/barcodes/static/src/js/barcode_form_view.js]: split into two lines for better visibility

**Comment by amoyaux** [comment]
@adr-odoo completed a test in enterprise since the only widgets that use notifychange = False are in stock_barcode. A second test that open a pop up and check that only one view listen, to the barcode would be great. However I couldn't find a test that open a view upper another, I would require your help on this one.

**Comment by amoyaux** [comment]
@adr-odoo and now?

File: addons/barcodes/static/tests/barcode_tests.js [hunk 2]
```diff
@@ -459,4 +459,82 @@ QUnit.test('barcode_scanned only trigger error for active view', function (asser
...
+
+    var delay = barcodeEvents.BarcodeEvents.max_time_between_keys_in_ms;
+    barcodeEvents.BarcodeEvents.max_time_between_keys_in_ms = 0;
+
+    this.data.order.onchanges = {
+        _barcode_scanned: function () {},
+    };
+
+    // Define a specific barcode_handler widget for this test case
+    var TestBarcodeHandler = AbstractField.extend({
```
Comment (adr-odoo) [inline addons/barcodes/static/tests/barcode_tests.js]: test?

**Pull request merged**